mat/svd.go

```diff
@@ -249,3 +259,117 @@ func (svd *SVD) VTo(dst *Dense) {
 	}
 	dst.Copy(tmp.T())
 }
+
+// SolveTo calculates the least-squares solution to a linear matrix equation,
+// AX = B, using a rank-truncated SVD based on cond. The result is placed in in dst.
+// It returns the residuals calculated from the complete SVD. For this value to be
+// valid the factorization must have been performed with at least SVDFullU.
+// The decomposition must have been factorized computing both the U and V singular
+// vectors, and dst must be either empty or have length equal to svd.Rank(cond).
+func (svd *SVD) SolveTo(dst *Dense, b Matrix, cond float64) []float64 {
```

Comment:
Wondering whether this should take `rank int` calculated from `svd.Rank(cond)`?

Reply:
I think so.

Also, note that it is possible to compute an approximate solution by taking a band of singular values. With full-rank matrices, it is a low-band. With rank-deficient matrices, it lies in the middle.

Reply:
In the case of rank 0, what do we leave in `dst`?

Reply:
I will just reset the `dst`.

Reply:
I believe that any matrix with content must have rank>=1. Rank 0 would be a matrix with all zeros.

The rank-1 approximation of the SVD is composed of the largest SV, and the rank-1 approximation of the LSE is composed of the smallest non-zero SV.

Reply:
Yes, that's true, but we still need to handle it. We can return doing nothing, panic, or something else. I've done something else, but I'm not sure if it's the right thing.

Reply:
I think a panic is fair. Any user computing the SVD of a matrix should first ensure that it contains information that can be extracted. For example, computing the variance of each column and removing the ones equal zero is a sensible pre-processing. By doing so, the user would find that the matrix is zero before calling SVD.

Doing something else might involve doing the check explicitly, but I think it would add overhead for users that have done their homework.

Reply:
Done

Reply:
FTR, Matlab returns a 0. Not a matrix. By a vector of singular values with a single element, a zero.